page-manager.js

```diff
@@ -672,11 +677,11 @@ define([
         _wrapControllerBeforeHandlingRoute : function(controller, controllerInfo, alreadyInStack) {
             if (controller.__wrapper) {
                 var wrapperOptions = ('string' === typeof controllerInfo.controllerWrapper) ?
-                                     {} : controllerInfo.controllerWrapper;
-
-                return new Promise(function(resolve) {
-                    controller.__wrapper.wrap(resolve, wrapperOptions, alreadyInStack);
-                });
+                                     {} : controllerInfo.controllerWrapper,
+                    wrapperPromise = new Promise(function(resolve) {
+                                        controller.__wrapper.wrap(resolve, wrapperOptions, alreadyInStack);
+                                    });
```

Comment:
I'll add it later, as I might need to make a little change to controller wrapper flow